git_sizer_test.go

```diff
@@ -24,6 +24,13 @@ import (
 	"github.com/github/git-sizer/sizes"
 )
 
+func TestMain(m *testing.M) {
+	// Allow Git to access the bare repositories of these tests, even if
+	// the user configured `safe.bareRepository=explicit` globally.
+	os.Setenv("GIT_CONFIG_PARAMETERS", "'safe.bareRepository=all'")
```

Comment:
This is not needed anymore, as the only time we were needing it was for the `git -C <path> rev-parse --git-dir` and the `cd <path> && git rev-parse --git-path shallow` calls, which are no longer called from the test suite for bare repositories.

This commit should therefore be dropped.

Reply:
Oh, that's a surprise. The test code _does_ create non-bare repositories, and is capable of processing them, but I guess it never does.

Maybe it should, to make sure that it works.

Reply:
The test code does create non-bare repositories, and everything (now) works as expected: the `git -C <path> rev-parse --git-dir` call works without any hitch there: https://github.com/github/git-sizer/blob/fb78b414e22c5c95dfb4c4847b6e7cee58b1b1af/git/git.go#L82

The problem I tried to paper over in e62a41c was that _bare_ repositories have a problem with that call if `safe.bareRepository=explicit` (which really should be the default, and I hope it will become the default soon).

I double-checked that that e62a41c could be dropped, but that's not so: the `git-sizer` invocations do not go through `repo.GitCommand()` and therefore have to set `GIT_DIR` explicitly, too. I did that in fb78b41.